aws/resource_aws_neptune_parameter_group.go

```diff
@@ -0,0 +1,280 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/neptune"
+)
+
+// We can only modify 20 parameters at a time, so walk them until
+// we've got them all.
+const maxParams = 20
+
+func resourceAwsNeptuneParameterGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNeptuneParameterGroupCreate,
+		Read:   resourceAwsNeptuneParameterGroupRead,
+		Update: resourceAwsNeptuneParameterGroupUpdate,
+		Delete: resourceAwsNeptuneParameterGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+				StateFunc: func(val interface{}) string {
+					return strings.ToLower(val.(string))
+				},
+			},
+			"family": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "Managed by Terraform",
+			},
+			"parameter": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"apply_method": {
```

Comment:
This argument was not documented originally and will be added post merge. Also adding validation:

```go
ValidateFunc: validation.StringInSlice([]string{
	neptune.ApplyMethodImmediate,
	neptune.ApplyMethodPendingReboot,
}, false),
```